### PR TITLE
Add a Clearing site data feature

### PR DESCRIPTION
**Pull request by Elchi3**
Draft until BCD fixes have been merged:
- https://github.com/mdn/browser-compat-data/pull/24821
- https://github.com/mdn/browser-compat-data/pull/24822

### CODE REVIEW

**Comment by Elchi3** [comment]
So, Clear-Site-Data is not fully interoperable. https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Clear-Site-Data#browser_compatibility. We could divide this in separate features:
- C-S-D: storage and cookies (the only parts that would be baseline)
- C-S-D: cache
- C-S-D: execution contexts
- C-S-D: clientHints

or we leave it as one feature and have a baseline false until C-S-D is more interoperable.

File: features/clear-site-data.yml [hunk 1]
```diff
@@ -1,6 +1,5 @@
-draft_date: 2024-10-28
-name: Clear Site Data
-description: TODO
+name: Clearing site data
+description: The `Clear-Site-Data header` deletes browsing data (cookies, storage, cache) associated with the requesting website.
```
Comment (ddbeck) [inline features/clear-site-data.yml:2]:
```suggestion
description: The `Clear-Site-Data` header deletes browsing data (such as cookies, storage, or cache) associated with the requesting website.
```

**Review by ddbeck — COMMENTED** [review]
Hmm, another way to look at this might be.

- A `Clear-Site-Data` feature consisting of `Clear-Site-Data`, `secure_context_required`, `storage`, `cookies`, and `wildcard`
- Put the others with their respective features?

I'd also be happy with wildcard being the iron core of this feature and pinning to it with `compute_from`.

**Comment by Elchi3** [comment]
I've pinned the wildcard support. I think that makes the most sense if we don't want to split out anything here for now.

**Review by ddbeck — APPROVED** [review]
This works for me. Thank you!

**Pull request merged**